azurerm/resource_arm_healthcare_service.go

```diff
@@ -0,0 +1,391 @@
+			AccessPolicies: expandAzureRMhealthcareapisAccessPolicyEntries(d),
+			CosmosDbConfiguration: &healthcareapis.ServiceCosmosDbConfigurationInfo{
+				OfferThroughput: &cdba,
+			},
+			CorsConfiguration:           expandAzureRMhealthcareapisCorsConfiguration(d),
+			AuthenticationConfiguration: expandAzureRMhealthcareapisAuthentication(d),
+		},
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resGroup, name, healthcareServiceDescription)
+	if err != nil {
+		return fmt.Errorf("Error Creating/Updating Healthcare Service %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error Creating/Updating Healthcare Service %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	read, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error Retrieving Healthcare Service %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Healthcare Service %q (resource group %q) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmHealthcareServiceRead(d, meta)
+}
+
+func resourceArmHealthcareServiceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).healthcare.HealthcareServiceClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["services"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[WARN] Healthcare Service %q was not found (Resource Group %q)", name, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Azure Healthcare Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if kind := resp.Kind; string(kind) != "" {
+		d.Set("kind", kind)
+	}
+	if properties := resp.Properties; properties != nil {
+		if config := properties.AccessPolicies; config != nil {
+			d.Set("access_policy_object_ids", flattenHealthcareAccessPolicies(config))
+		}
+		if config := properties.CosmosDbConfiguration; config != nil {
+			d.Set("cosmosdb_throughput", config.OfferThroughput)
+		}
+
+		authOutput := make([]interface{}, 0)
+		if authConfig := properties.AuthenticationConfiguration; authConfig != nil {
+			output := make(map[string]interface{})
+			if authConfig.Authority != nil {
+				output["authority"] = *authConfig.Authority
+			}
+			if authConfig.Audience != nil {
+				output["audience"] = *authConfig.Audience
+			}
+			if authConfig.SmartProxyEnabled != nil {
+				output["smart_proxy_enabled"] = *authConfig.SmartProxyEnabled
+			}
+			authOutput = append(authOutput, output)
+		}
```

Comment:
Could we make this into a flatten function?

Reply:
done